src/hotspot/share/opto/vectorization.hpp

```diff
@@ -445,6 +451,39 @@ class VLoopTypes : public StackObj {
   const Type* container_type(Node* n) const;
 };
 
+// Submodule of VLoopAnalyzer.
+// We compute and cache the VPointer for every load and store.
+class VLoopPointers : public StackObj {
+private:
+  Arena*                   _arena;
+  const VLoop&             _vloop;
+  const VLoopBody&         _body;
+
+  // Array of cached pointers
```

Comment:
Maybe make a note that we allocate/cache them lazily upon request.

Reply:
It is not lazy, they are allocated and cached in `compute_and_cache`. Like all other `VLoopAnalyzer` submodules. Maybe I missed your point :sweat_smile:

Reply:
I've meant that it's not allocated in the constructor as you initialize it with `nullptr`. It's only initialized once you call `compute_and_cache()` which may not happen if we bail out earlier. That's what I've meant with "lazy" but that was probably not clear enough :-)

Reply:
Aha, I see. I mean all other submodules are handled the same. They also cannot really be used until `VLoopAnalyzer::setup_submodules` returns with success. I guess this here is the first instance where the data structure itself is only allocated after the constructor. But I feel like if anybody has a question about where it is allocated, they can just search the reference. If I start putting down such detailed comments, then I need to put them everywhere. And that will clutter the code.

Reply:
That's true. Here I think I've only commented it since it's allocated specially for the first time in the sub modules. But it does not really add much information per se. It's fine to leave it like that.

Reply:
Ok, thanks for the suggestion anyway :blush: 
I will leave it without a comment then.